### ISSUE TITLE
Cell-Broadcast support? (used for regional emergency alerts)

### ISSUE THREAD

**Issue by barrystaes** [signalapp/Signal-Android#4950]
The "Cell-Broadcast" ([SMS-CB](https://en.wikipedia.org/wiki/Cell_Broadcast)) is not a regular SMS, but usually the SMS application is used to show it. Its used by the Dutch national/regional emergency alert system [NLAlert](http://www.nl-alert.nl/over-nl-alert/) since 2012. They run a yearly emergency test.

However with Signal now installed, i encountered a problem while trying to see their test message. I put a [screenshot on Twitter](https://twitter.com/BarryStaes/status/673823993787498496). This problem is hard to reproduce, there is only one test a year. All the more reason to help by gathering relevant info on this issue here..

**Details on the problem i encountered:**
I received this test OK the years before using Signal on my device. This time, several things happened;
- My phone did receive the Cell-Broadcast
- My phone did notify me that there was a new message
- Tapping the notification showed me the Samsung stock SMS app. It has an error message that translates to "Messages has stopped". (left in twitter screenshot)
- When dismissing that, the next screen now has a dialog forcing me to set this app to default. (right in twitter screenshot)
- When confirming that, the message is presented.
- Inside the Messages app, its stored in a conversation that translates to "Infomessages", with a cell tower icon.
- Signal was no longer my default SMS app, i had to configure that again.

**TLDR / conclusion:**
- Good news is, i was at least _able to_ read the national emergency test message. :+1:
- Bad news is, some people might not be able to maneuver this minor technical hurdle, therefor not reading the emergency message. :fearful:

**IMHO there are two problems here:**
- Signal seems to have no support for the emergency Cell-Broadcast.
  Can anyone confirm?
- Samsung should not have their Messages app crash and require being set as default SMS app, before showing me the emergency cell-broadcast.

**Comment by darryl-grennan** [comment]
Curious as to what other SMS messages are handled by stock messengers but not Signal.
Does Signal support the following:
https://en.wikipedia.org/wiki/Short_Message_Service#Flash_SMS
https://en.wikipedia.org/wiki/Short_Message_Service#Silent_SMS

The "Flash SMS" being the one I think barrystaes is talking about.

Interesting read here:
https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/wiki/Special-SMS

**Comment by barrystaes** [comment]
I refer to SMS-CB:
https://en.wikipedia.org/wiki/Cell_Broadcast
> Cell Broadcast is a mobile technology that allows messages (up to 15 pages of up to 93 characters) to be broadcast to all mobile handsets and similar devices within a designated geographical area. The broadcast range can be varied, from a single cell to the entire network.

**Comment by johanw666** [comment]
Yes, I have issues with that too. My Sony Z1 Compct doesn't receive them at all (not even on the stock sms app), although it SHOULD work since my provider claimed they had it working on 4G now.

BTW, AFAIK the Dutch emergency alert system tests this the first of every month, when those alarms go off as well.

**Comment by InfinityLF** [comment]
CyanogenMOD 12.1 (latest daily back then) handled the Cell Broadcast perfectly by showing a system alert. In my opinion this is something for the ROM to take care of, not for Signal.

**Comment by barrystaes** [comment]
I dont believe a system alert is "re-readable" / findable afte dismissing it, and therefor i think this approach is rather inadequate for showing SMS-CB emergency text messages.

**Comment by InfinityLF** [comment]
I thought the message was kept in the Cell Broadcasts app, but I'm not sure anymore. Due to a fresh install of CM13 I can't look it up and NLAlert doesn't seem to have a testing service.

This is assuming that Cell Broadcasts is a different app than the default/CM messaging app. The recent apps switcher seems to think so too. Nevertheless, it is quite hard to find: Messaging app -> Settings -> Advanced -> Wireless Alerts. Perhaps that's the reason Samsung has put the broadcasts in the Messaging app itself?

**Comment by nkempe** [comment]
This is an old issue but still present.
As mentioned before, in the Netherlands we receive emergency alerts via cell broadcasts, so this is an important issue in that respect. By 2020 the old sirens will be removed so we will probably be dependant on broadcast messages by then.
Last Monday August 5th another test broadcast was sent, and my phone didn't do anything. Before Signal I was able to receive the messages.

FYI: the test broadcasts are send twice a year, the next one will be on June 5th 2017.
The broadcasts happen on channel 919, according to the Dutch info on http://www.crisis.nl/nl-alert/instelhulp/fairphone.aspx
Same site in Google Translate: https://translate.google.com/translate?sl=nl&tl=en&js=y&prev=_t&hl=en&ie=UTF-8&u=http%3A%2F%2Fwww.crisis.nl%2Fnl-alert%2Finstelhulp%2Ffairphone.aspx&edit-text=

**Comment by johanw666** [comment]
This is pretty vagely documented as far as I can find out. This is the most informative posting I could find on the subject: http://stackoverflow.com/questions/7118378/how-to-get-cell-broadcast-message . I'll try to insert that code into Signal or Silence and see what happens.

 I tried switching my phone to the standard message app before I knew a cell broadcast was going to arrive, but no success anyway on a Z1 compact. A cheap Acer Z205 which I use mainly as road navigator and test phone did receive the Dutch alert message though.

**Comment by kb100** [comment]
This happens for me with Emergency alerts (from: "Emergency Alerts") and with push messages from T-Mobile about data usage (from: "889"). Would be great if Signal could display these.

**Comment by xurizaemon** [comment]
NZ ran a nationwide test of this functionality today. I saw the OS notification, but was not being able to read the alert text via SMS app after the fact (and saw similar reports from other Android/Signal users on Twitter). It seems this is the intended behaviour, that the SMS client would display the message so that users in emergency situations can act on details sent via emergency notification after the initial broadcast.

* [example tweets posted at the time](https://twitter.com/xurizaemon/status/934668021372719105)

**Comment by johanw666** [comment]
After some study I found out that these kind of messages are technically not an sms message so neither Signal or the stock sms app will  receive them unless the stock app has some extra functionality. On my Sony Z3 comoact I have a special app for it, on the old Z1 I have also installed it but it has to be installed as a system app.

**Comment by automated-signal** [comment]
GitHub Issue Cleanup:
See #7598 for more information.
